### PR TITLE
[Data][Docs] Revise "ML Tensor Support"

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

Users are confused by the inconsistent use of "tensor" and "array" in the Ray Data documentation. This PR clarifies that tensor data is represented as ndarrays.

In addition, this PR revises the "ML Tensor Support" user guide to remove redundant or outdated information.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/data/working-with-ndarrays.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
+.. _working_with_ndarrays:
```
Comment (amogkam) [inline doc/source/data/working-with-ndarrays.rst:1]: this needs to be added in the table of contents somewhere?

File: doc/source/data/data.rst [hunk 2]
```diff
@@ -44,16 +44,16 @@ As part of the Ray ecosystem, Ray Data can leverage the full functionality of Ra
...
 :ref:`Data User Guide <data_user_guide>`.
 
 ---------------------------------------
 Streaming Preprocessing for ML Training
 ---------------------------------------
 
 Use Ray Data to load and preprocess data for distributed :ref:`ML training pipelines <train-docs>` in a streaming fashion.
-Ray Data is intended to serve as a last-mile bridge from storage or ETL pipeline outputs to distributed 
-applications and libraries in Ray. Don't use it as a replacement for more general data 
+Ray Data is intended to serve as a last-mile bridge from storage or ETL pipeline outputs to distributed
```
Comment (angelinalg) [inline doc/source/data/data.rst:55]:
```suggestion
Ray Data serves as a last-mile bridge from storage or ETL pipeline outputs to distributed
```

File: doc/source/data/key-concepts.rst [hunk 3]
```diff
@@ -100,7 +101,7 @@ Fault tolerance
 
 Datastream performs *lineage reconstruction* to recover data. If an application error or
 system failure occurs, Datastream recreates lost blocks by re-executing tasks. If ``compute=ActorPoolStrategy(size=n)`` is used, then Ray
-will restart the actor used for computing the block prior to re-executing the task. 
+will restart the actor used for computing the block prior to re-executing the task.
```
Comment (angelinalg) [inline doc/source/data/key-concepts.rst:104]:
```suggestion
restarts the actor used for computing the block prior to re-executing the task.
```

File: doc/source/data/working-with-ndarrays.rst (hunk 1 above)
Comment (bveeramani) [reply doc/source/data/working-with-ndarrays.rst:1]: It's in `user-guider.rst`.

**Review by ericl — APPROVED** [review]

**Comment by bveeramani** [comment]
Failing tests seem to be

```
ImportError: cannot import name 'GcsPublisher' from 'ray._private.gcs_pubsub' (/ray/python/ray/_private/gcs_pubsub.py)
```

**Pull request merged**